tasks/release.py

```diff
@@ -1022,3 +900,408 @@ def tag_version(ctx, agent_version, commit="HEAD", verify=True, push=True, force
+                "red",
+            ),
+            code=1,
+        )
+
+    # Find milestone based on what the next final version is. If the milestone does not exist, fail.
+    milestone_name = str(next_final_version(ctx, max(list_major_versions)))
+
+    milestone = github.get_milestone_by_name(milestone_name)
+
+    if not milestone or not milestone["number"]:
+        raise Exit(
+            color_message(
+                """Could not find milestone {} in the Github repository. Response: {}
+Make sure that milestone is open before trying again.""".format(
+                    milestone_name, milestone
+                ),
+                "red",
+            ),
+            code=1,
+        )
+
+    # Step 1: Update release entries
+
+    print(color_message("Updating release entries", "bold"))
+    for major_version in list_major_versions:
+        new_version = next_rc_version(ctx, major_version, patch_version)
+        update_release_json(github.api_token, new_version)
+
+    # Step 2: Update internal module dependencies
+
+    print(color_message("Updating Go modules", "bold"))
+    update_modules(ctx, str(new_highest_version))
+
+    # Step 3: branch out, commit change, push branch
+
+    print(color_message("Branching out to {}".format(update_branch), "bold"))
+    ctx.run("git checkout -b {}".format(update_branch))
+
+    print(color_message("Committing release.json and Go modules updates", "bold"))
+    print(
+        color_message(
+            "If commit signing is enabled, you will have to make sure the commit gets properly signed.", "bold"
+        )
+    )
+    ctx.run("git add release.json")
+    ctx.run("git ls-files . | grep 'go.mod$' | xargs git add")
+
+    res = ctx.run("git commit -m 'Update release.json and Go modules for {}'".format(versions_string), warn=True)
+    if res.exited is None or res.exited > 0:
+        raise Exit(
+            color_message(
+                "Could not create commit. Please commit manually, push the {} branch and then open a PR against {}.".format(
+                    update_branch,
+                    current_branch,
+                ),
+                "red",
+            ),
+            code=1,
+        )
+
+    print(color_message("Pushing new branch to the upstream repository", "bold"))
+    res = ctx.run("git push --set-upstream {} {}".format(upstream, update_branch), warn=True)
+    if res.exited is None or res.exited > 0:
+        raise Exit(
+            color_message(
+                "Could not push branch {} to the upstream '{}'. Please push it manually and then open a PR against {}.".format(
+                    update_branch,
+                    upstream,
+                    current_branch,
+                ),
+                "red",
+            ),
+            code=1,
+        )
+
+    print(color_message("Creating PR", "bold"))
+
+    # Step 4: create PR
+
+    pr = github.create_pr(
```

Comment:
I got to this PR bit late but it may be useful to do a draft PR instead of a regular PR for this to prevent repo noise when someone is checking/working on the tooling or just have an extra level of assurance that it wasn't an accidental invocation by the author. It's fine as-is for sure but maybe something to consider if it seems useful.